### ISSUE TITLE
Possible bug in TensorFlowInferenceInterface

### ISSUE THREAD

**Issue by singlasahil14** [tensorflow/tensorflow#9694]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux 16.04
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.1.0rc2
- **Bazel version (if compiling from source)**: 
- **CUDA/cuDNN version**: Not using
- **GPU model and memory**: Not using
- **Exact command to reproduce**: Need to run tensorflow code in Android

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.
I was testing a trained my style transfer model on android. It uses depthwise separable convolution to decrease the size of neural network. When working on square images, it gives good results. But when I work with it on rectangular images, it gives weird results. Please see the result images attached. I highly doubt that this is due to my code because it is giving good results on all kinds of images on my laptop.
The style image was the famous painting 'la muse'
Content image:
![content](https://cloud.githubusercontent.com/assets/6660192/25755856/3ad8a43c-31e2-11e7-91f7-9758177f2002.jpg)
Result image for rectangular input:
![rectangular_image_result](https://cloud.githubusercontent.com/assets/6660192/25755900/600d2638-31e2-11e7-9593-890a42360f61.png)
Result image for square input:
![square_image_result](https://cloud.githubusercontent.com/assets/6660192/25755899/60060d8a-31e2-11e7-89c2-ce554c831e7c.png)
 
### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.
This is the code I used for running the network on android using TensorflowInferenceInterface
  private void stylizeImage(final Bitmap bitmap) {
    bitmap.getPixels(intValues, 0, bitmap.getWidth(), 0, 0, bitmap.getWidth(), bitmap.getHeight());

    for (int i = 0; i < intValues.length; ++i) {
      final int val = intValues[i];
      floatValues[i * 3] = ((val >> 16) & 0xFF) / 255.0f;
      floatValues[i * 3 + 1] = ((val >> 8) & 0xFF) / 255.0f;
      floatValues[i * 3 + 2] = (val & 0xFF) / 255.0f;
    }

    inferenceInterface.feed(
        INPUT_NODE, floatValues, 1, bitmap.getWidth(), bitmap.getHeight(), 3);

    inferenceInterface.run(new String[] {OUTPUT_NODE}, false);
    inferenceInterface.fetch(OUTPUT_NODE, outValues);

    for (int i = 0; i < intValues.length; ++i) {
      intValues[i] =
          0xFF000000
              | ((outValues[i * 3]) << 16)
              | ((outValues[i * 3 + 1]) << 8)
              | (outValues[i * 3 + 2]);
    }

    bitmap.setPixels(intValues, 0, bitmap.getWidth(), 0, 0, bitmap.getWidth(), bitmap.getHeight());
  }

**Comment by andrewharp** [comment]
@keveman This sort of behavior is expected when using non-square inputs, correct?

@singlasahil14 As a workaround, you might try padding the image on top and bottom so that it is square, stylizing, and then re-cropping.

**Comment by Androbin** [comment]
Could it be that you mixed up the image width and height? For example, the common format is [batch, height, width, depth], so you may want to check that.

**Comment by andrewharp** [comment]
Closing due to lack of activity; please respond with additional info if the issue persists.
